File: source/aoBlogs2/Models/View/RequestModel.vb
Imports Contensive.BaseClasses

Namespace Models.View
    Public Class RequestModel
        '
        '====================================================================================================
        '
        Private ReadOnly cp As CPBaseClass
        '
        '
        Public Sub New(cp As CPBaseClass)
            Me.cp = cp

        End Sub
        '
        '
        Public ReadOnly Property page As Integer
            Get
                Return cp.Doc.GetInteger(rnPageNumber)
            End Get
        End Property
        '
        '
        Public ReadOnly Property instanceGuid As String
            Get
                instanceGuid = cp.Doc.GetText("instanceId")
                If (String.IsNullOrWhiteSpace(instanceGuid)) Then instanceGuid = "BlogWithoutinstanceId-PageId-" & cp.Doc.PageId
                Return instanceGuid
            End Get
        End Property
        '
        '
        Public ReadOnly Property srcViewId As Integer
            Get
                Return cp.Doc.GetInteger(rnFormID)
            End Get
        End Property
        '
        '
        Public Property dstViewId As Integer
        '
        '====================================================================================================
        ''' <summary>
        ''' category Id
        ''' </summary>
        ''' <returns></returns>
        Public ReadOnly Property categoryId As Integer
            Get
                If (_categoryId Is Nothing) Then
                    If (Not String.IsNullOrEmpty(cp.Doc.GetText(RequestNameBlogCategoryIDSet))) Then
                        _categoryId = cp.Doc.GetInteger(RequestNameBlogCategoryIDSet)
                    Else
                        _categoryId = cp.Doc.GetInteger(RequestNameBlogCategoryID)
                    End If
                End If
                Return Convert.ToInt32(_categoryId)
            End Get
        End Property
        Private _categoryId As Integer? = Nothing
        '
        '====================================================================================================
        ''' <summary>
        ''' The Blog Entry Id requested
        ''' </summary>
        ''' <returns></returns>
        Public ReadOnly Property blogEntryId As Integer
            Get
                If (_blogEntryId Is Nothing) Then
                    _blogEntryId = cp.Doc.GetInteger(RequestNameBlogEntryID)
                End If
                Return Convert.ToInt32(_blogEntryId)
            End Get
        End Property
        Private _blogEntryId As Integer? = Nothing
        '
        '====================================================================================================
        '
        Public ReadOnly Property ButtonValue As String
            Get
                If (_ButtonValue Is Nothing) Then
                    _ButtonValue = cp.Doc.GetText("button")
                End If
                Return Convert.ToString(_ButtonValue)
            End Get
        End Property
        Private _ButtonValue As String = Nothing
        '
        '====================================================================================================
        '
        Public ReadOnly Property FormID As Integer
            Get
                If (_FormID Is Nothing) Then
                    _FormID = cp.Doc.GetInteger(rnFormID)
                End If
                Return Convert.ToInt32(_FormID)
            End Get
        End Property
        Private _FormID As Integer? = Nothing

        '
        '====================================================================================================
        '
        Public ReadOnly Property SourceFormID As Integer
            Get
                If (_SourceFormID Is Nothing) Then
                    _SourceFormID = cp.Doc.GetInteger(RequestNameSourceFormID)
                End If
                Return Convert.ToInt32(_SourceFormID)
            End Get
        End Property
        Private _SourceFormID As Integer? = Nothing
        '
        '====================================================================================================
        '
        Public ReadOnly Property KeywordList As String
            Get
                If (_KeywordList Is Nothing) Then
                    _KeywordList = cp.Doc.GetText(RequestNameKeywordList)
                End If
                Return Convert.ToString(_KeywordList)
            End Get
        End Property
        Private _KeywordList As String = Nothing
        '
        '====================================================================================================
        '
        Public ReadOnly Property DateSearchText As String
            Get
                If (_DateSearchText Is Nothing) Then
                    _DateSearchText = cp.Doc.GetText(RequestNameDateSearch)
                End If
                Return Convert.ToString(_DateSearchText)
            End Get
        End Property
        Private _DateSearchText As String = Nothing
        '
        '====================================================================================================
        '
        Public ReadOnly Property ArchiveMonth As Integer
            Get
                If (_ArchiveMonth Is Nothing) Then
                    _ArchiveMonth = cp.Doc.GetInteger(RequestNameArchiveMonth)
                End If
                Return Convert.ToInt32(_ArchiveMonth)
            End Get
        End Property
        Private _ArchiveMonth As Integer? = Nothing
        '
        '====================================================================================================
        '
        Public ReadOnly Property ArchiveYear As Integer
            Get
                If (_ArchiveYear Is Nothing) Then
                    _ArchiveYear = cp.Doc.GetInteger(RequestNameArchiveYear)
                End If
                Return Convert.ToInt32(_ArchiveYear)
            End Get
        End Property
        Private _ArchiveYear As Integer? = Nothing
        '
        '====================================================================================================
        '
        Public ReadOnly Property EntryID As Integer
            Get
                If (_EntryID Is Nothing) Then
                    _EntryID = cp.Doc.GetInteger(RequestNameBlogEntryID)
                End If
                Return Convert.ToInt32(_EntryID)
            End Get
        End Property
        Private _EntryID As Integer? = Nothing
        '
        '====================================================================================================
        '
        Public ReadOnly Property BlogEntryName As String
            Get
                If (_BlogEntryName Is Nothing) Then
                    _BlogEntryName = cp.Doc.GetText(RequestNameBlogEntryName)
                End If
                Return Convert.ToString(_BlogEntryName)
            End Get
        End Property
        Private _BlogEntryName As String = Nothing
        '
        '====================================================================================================
        '
        Public ReadOnly Property BlogEntryCopy As String
            Get
                If (_BlogEntryCopy Is Nothing) Then
                    _BlogEntryCopy = cp.Doc.GetText(RequestNameBlogEntryCopy)
                End If
                Return Convert.ToString(_BlogEntryCopy)
            End Get
        End Property
        Private _BlogEntryCopy As String = Nothing
        '
        '====================================================================================================
        '
        Public ReadOnly Property BlogEntryTagList As String
            Get
                If (_BlogEntryTagList Is Nothing) Then
                    _BlogEntryTagList = cp.Doc.GetText(RequestNameBlogEntryTagList)
                End If
                Return Convert.ToString(_BlogEntryTagList)
            End Get
        End Property
        Private _BlogEntryTagList As String = Nothing
        '
        '====================================================================================================
        '
        Public ReadOnly Property BlogEntryCategoryId As Integer
            Get
                If (_BlogEntryCategoryId Is Nothing) Then
                    _BlogEntryCategoryId = cp.Doc.GetInteger(RequestNameBlogEntryCategoryID)
                End If
                Return Convert.ToInt32(_BlogEntryCategoryId)
            End Get
        End Property
        Private _BlogEntryCategoryId As Integer? = Nothing
    End Class
End Namespace

File: source/aoBlogs2/Models/Domain/RequestModel.cs
using System;
using Contensive.BaseClasses;

namespace Contensive.Addons.Blog.Models.View {
    public class RequestModel {
        // 
        // ====================================================================================================
        // 
        private readonly CPBaseClass cp;
        // 
        // 
        public RequestModel(CPBaseClass cp) {
            this.cp = cp;

        }
        // 
        // 
        public int page {
            get {
                return cp.Doc.GetInteger(constants.rnPageNumber);
            }
        }
        // 
        // 
        public string instanceGuid {
            get {
                string instanceGuidRet = default;
                instanceGuidRet = cp.Doc.GetText("instanceId");
                if (string.IsNullOrWhiteSpace(instanceGuidRet))
                    instanceGuidRet = "BlogWithoutinstanceId-PageId-" + cp.Doc.PageId;
                return instanceGuidRet;
            }
        }
        // 
        // 
        public int srcViewId {
            get {
                return cp.Doc.GetInteger(constants.rnFormID);
            }
        }
        // 
        // 
        public int dstViewId { get; set; }
        // 
        // ====================================================================================================
        /// <summary>
        /// category Id
        /// </summary>
        /// <returns></returns>
        public int categoryId {
            get {
                if (_categoryId is null) {
                    if (!string.IsNullOrEmpty(cp.Doc.GetText(constants.RequestNameBlogCategoryIDSet))) {
                        _categoryId = cp.Doc.GetInteger(constants.RequestNameBlogCategoryIDSet);
                    }
                    else {
                        _categoryId = cp.Doc.GetInteger(constants.RequestNameBlogCategoryID);
                    }
                }
                return Convert.ToInt32(_categoryId);
            }
        }
        private int? _categoryId = default;
        // 
        // ====================================================================================================
        /// <summary>
        /// The Blog Entry Id requested
        /// </summary>
        /// <returns></returns>
        public int blogEntryId {
            get {
                if (_blogEntryId is null) {
                    _blogEntryId = cp.Doc.GetInteger(constants.RequestNameBlogEntryID);
                }
                return Convert.ToInt32(_blogEntryId);
            }
        }
        private int? _blogEntryId = default;
        // 
        // ====================================================================================================
        // 
        public string ButtonValue {
            get {
                if (_ButtonValue is null) {
                    _ButtonValue = cp.Doc.GetText("button");
                }
                return Convert.ToString(_ButtonValue);
            }
        }
        private string _ButtonValue = null;
        // 
        // ====================================================================================================
        // 
        public int FormID {
            get {
                if (_FormID is null) {
                    _FormID = cp.Doc.GetInteger(constants.rnFormID);
                }
                return Convert.ToInt32(_FormID);
            }
        }
        private int? _FormID = default;

        // 
        // ====================================================================================================
        // 
        public int SourceFormID {
            get {
                if (_SourceFormID is null) {
                    _SourceFormID = cp.Doc.GetInteger(constants.RequestNameSourceFormID);
                }
                return Convert.ToInt32(_SourceFormID);
            }
        }
        private int? _SourceFormID = default;
        // 
        // ====================================================================================================
        // 
        public string KeywordList {
            get {
                if (_KeywordList is null) {
                    _KeywordList = cp.Doc.GetText(constants.RequestNameKeywordList);
                }
                return Convert.ToString(_KeywordList);
            }
        }
        private string _KeywordList = null;
        // 
        // ====================================================================================================
        // 
        public string DateSearchText {
            get {
                if (_DateSearchText is null) {
                    _DateSearchText = cp.Doc.GetText(constants.RequestNameDateSearch);
                }
                return Convert.ToString(_DateSearchText);
            }
        }
        private string _DateSearchText = null;
        // 
        // ====================================================================================================
        // 
        public int ArchiveMonth {
            get {
                if (_ArchiveMonth is null) {
                    _ArchiveMonth = cp.Doc.GetInteger(constants.RequestNameArchiveMonth);
                }
                return Convert.ToInt32(_ArchiveMonth);
            }
        }
        private int? _ArchiveMonth = default;
        // 
        // ====================================================================================================
        // 
        public int ArchiveYear {
            get {
                if (_ArchiveYear is null) {
                    _ArchiveYear = cp.Doc.GetInteger(constants.RequestNameArchiveYear);
                }
                return Convert.ToInt32(_ArchiveYear);
            }
        }
        private int? _ArchiveYear = default;
        // 
        // ====================================================================================================
        // 
        public int EntryID {
            get {
                if (_EntryID is null) {
                    _EntryID = cp.Doc.GetInteger(constants.RequestNameBlogEntryID);
                }
                return Convert.ToInt32(_EntryID);
            }
        }
        private int? _EntryID = default;
        // 
        // ====================================================================================================
        // 
        public string BlogEntryName {
            get {
                if (_BlogEntryName is null) {
                    _BlogEntryName = cp.Doc.GetText(constants.RequestNameBlogEntryName);
                }
                return Convert.ToString(_BlogEntryName);
            }
        }
        private string _BlogEntryName = null;
        // 
        // ====================================================================================================
        // 
        public string BlogEntryCopy {
            get {
                if (_BlogEntryCopy is null) {
                    _BlogEntryCopy = cp.Doc.GetText(constants.RequestNameBlogEntryCopy);
                }
                return Convert.ToString(_BlogEntryCopy);
            }
        }
        private string _BlogEntryCopy = null;
        // 
        // ====================================================================================================
        // 
        public string BlogEntryTagList {
            get {
                if (_BlogEntryTagList is null) {
                    _BlogEntryTagList = cp.Doc.GetText(constants.RequestNameBlogEntryTagList);
                }
                return Convert.ToString(_BlogEntryTagList);
            }
        }
        private string _BlogEntryTagList = null;
        // 
        // ====================================================================================================
        // 
        public int BlogEntryCategoryId {
            get {
                if (_BlogEntryCategoryId is null) {
                    _BlogEntryCategoryId = cp.Doc.GetInteger(constants.RequestNameBlogEntryCategoryID);
                }
                return Convert.ToInt32(_BlogEntryCategoryId);
            }
        }
        private int? _BlogEntryCategoryId = default;
    }
}